static/scripts/rewards/gift-cards/gift-card.ts

```diff
@@ -0,0 +1,116 @@
+import { BigNumberish } from "ethers";
+import { GiftCard } from "../../../../shared/types";
+import { getFixedPriceToValueMap, getGiftCardValue, getRangePriceToValueMap, isRangePriceGiftCardClaimable } from "../../../../shared/pricing";
+import { formatEther } from "ethers/lib/utils";
+
+const html = String.raw;
+
+export function getGiftCardHtml(giftCard: GiftCard, allowBuy: boolean, rewardAmount: BigNumberish) {
+  return html`
```

Comment:
This syntax is wild

Reply:
True.  I am going to try to refactor this.

Reply:
I have been able to get rid of one argument out of function. After this, the function looks ok I think
`String.raw` is not crucial in the code. It is to make prettier formate the HTML inside template literals. We can remove it if you want but then HTML inside template literals will not be formatted by prettier.